### DISCUSSION TITLE
Is there have osu beatmap code for developing&editing?

### DISCUSSION THREAD

**Discussion by Toyeonjeangikki** [ppy/osu#14706] [Q&A] [answered]
I want to edit the osu beatmap option like, changing the hp drain decrease rate or etc.
Is there have osu beatmap code in github. I can only find osu lazer ans thats not what im looking for

**Comment by smoogipoo** [comment, marked as answer]
Presumably you mean this: https://osu.ppy.sh/wiki/en/osu%21_File_Formats/Osu_%28file_format%29

**Reply by Toyeonjeangikki** [reply]
Is isn’t a Osu lazer?

**Reply by smoogipoo** [reply]
I don't understand what you're looking for.
